x-pack/plugins/data_visualizer/public/application/index_data_visualizer/search_strategy/requests/get_document_stats.ts

```diff
@@ -12,20 +12,98 @@ import { DataPublicPluginStart, ISearchOptions } from '@kbn/data-plugin/public';
 import seedrandom from 'seedrandom';
 import { isDefined } from '@kbn/ml-is-defined';
 import { buildBaseFilterCriteria } from '@kbn/ml-query-utils';
+import { lastValueFrom } from 'rxjs';
+import { AggregateQuery } from '@kbn/es-query';
 import { RANDOM_SAMPLER_PROBABILITIES } from '../../constants/random_sampler';
 import type {
   DocumentCountStats,
   OverallStatsSearchStrategyParams,
 } from '../../../../../common/types/field_stats';
+import { escapeESQL, isESQLQuery } from './esql_utils';
 
 const MINIMUM_RANDOM_SAMPLER_DOC_COUNT = 100000;
 const DEFAULT_INITIAL_RANDOM_SAMPLER_PROBABILITY = 0.000001;
 
+export const getESQLDocumentCountStats = async (
+  search: DataPublicPluginStart['search'],
+  query: AggregateQuery,
+  filter?: estypes.QueryDslQueryContainer,
+  timeFieldName?: string,
+  intervalMs?: number,
+  searchOptions?: ISearchOptions
+): Promise<{ documentCountStats?: DocumentCountStats; totalCount: number }> => {
+  if (!isESQLQuery(query)) {
+    throw Error('No ESQL query provided');
+  }
+  const esqlBaseQuery = query.esql;
+  let earliestMs = Infinity;
+  let latestMs = -Infinity;
+
+  if (timeFieldName) {
+    const aggQuery = ` | EVAL _timestamp_= TO_DOUBLE(DATE_TRUNC(${intervalMs} millisecond, ${escapeESQL(
+      timeFieldName
+    )}))
+    | stats rows = count(*) by _timestamp_
+    | LIMIT 10000`;
+
+    const esqlResults = await lastValueFrom(
+      search.search(
+        {
+          params: {
+            query: esqlBaseQuery + aggQuery,
+            ...(filter ? { filter } : {}),
+          },
+        },
+        { ...(searchOptions ?? {}), strategy: 'esql' }
+      )
+    );
+
+    let totalCount = 0;
+    const _buckets: Record<string, number> = {};
+    // @ts-expect-error ES types needs to be updated with columns and values as part of esql response
+    esqlResults.rawResponse.values.forEach((val) => {
+      const [count, bucket] = val;
+      _buckets[bucket] = count;
+      totalCount += count;
+      if (bucket < earliestMs) {
+        earliestMs = bucket;
+      }
+      if (bucket >= latestMs) {
+        latestMs = bucket;
+      }
+    });
+    const result: DocumentCountStats = {
+      interval: intervalMs,
+      probability: 1,
+      randomlySampled: false,
+      timeRangeEarliest: earliestMs,
```

Comment:
I'm seeing some odd behavior with the 'Use full data' button with wildcards in the `from` index. For example with cloudwatch data, here it thinks there is only data on one day.

<img width="1366" alt="image" src="https://github.com/elastic/kibana/assets/7405507/fd752114-e91c-4173-b29a-abf9b910abed">

Possibly related to the fact that the first cloudwatch style data view I had created was for `cloudwatch-2019.11.04`. If I delete that data view, it all works as expected.